iavl_nodedb.go

```diff
@@ -147,27 +147,29 @@ func (ndb *nodeDB) Has(hash []byte) bool {
 //
 // Note that this function clears leftNode/rigthNode recursively and calls
 // hashWithCount on the given node.
-func (ndb *nodeDB) SaveBranch(node *IAVLNode, cb func(*IAVLNode)) {
+func (ndb *nodeDB) SaveBranch(node *IAVLNode, cb func(*IAVLNode)) []byte {
```

Comment:
Nice. I like this method.
And I think the return value is quite useful, as that hash is our persistent identifier

Reply:
And one recursive tree traversal is better

Reply:
:tada: